src/main/scala/TableGroup.scala

```diff
@@ -1,6 +1,412 @@
+            warnings :+= ErrorMessage(
+              "invalid_property",
+              "metadata",
+              "",
+              "",
+              property,
+              ""
+            )
+          }
+        }
+      }
+      ()
+    }
+    (annotations, commonProperties, inheritedProperties, warnings)
+  }
+
+  private def createTableObjectsAndSetWarnings(
+      tableGroupNode: ObjectNode,
+      baseUrl: String,
+      lang: String,
+      commonProperties: Map[String, JsonNode],
+      inheritedProperties: Map[String, JsonNode]
+  ): (Map[String, Table], Array[ErrorMessage]) = {
+    var warnings = Array[ErrorMessage]()
+
+    tableGroupNode.path("tables") match {
+      case t: ArrayNode if t.isEmpty() =>
+        throw new MetadataError("Empty tables property")
+      case t: ArrayNode =>
+        extractTablesFromNode(
+          t,
+          baseUrl,
+          lang,
+          commonProperties,
+          inheritedProperties
+        )
+      case n if n.isMissingNode => throw new MetadataError("No tables property")
+      case _                    => throw new MetadataError("Tables property is not an array")
+    }
+  }
+
+  def extractTablesFromNode(
+      tablesArrayNode: ArrayNode,
+      baseUrl: String,
+      lang: String,
+      commonProperties: Map[String, JsonNode],
+      inheritedProperties: Map[String, JsonNode]
+  ): (mutable.Map[String, Table], Array[ErrorMessage]) = {
+    var warnings = Array[ErrorMessage]()
+    val tables = Map[String, Table]()
+    for (tableElement <- tablesArrayNode.elements().asScalaArray) {
+      tableElement match {
+        case tableElementObject: ObjectNode => {
+          var tableUrl = tableElement.get("url")
+          if (!tableUrl.isTextual) {
+            warnings = warnings :+ ErrorMessage(
+              "invalid_url",
+              "metadata",
+              "",
+              "",
+              s"url: $tableUrl",
+              ""
+            )
+            tableUrl = new TextNode("")
+          }
+          tableUrl = new TextNode(
+            new URL(new URL(baseUrl), tableUrl.asText()).toString
+          )
+          tableElementObject.set("url", tableUrl)
+          val table = Table.fromJson(
+            tableElementObject,
+            baseUrl,
+            lang,
+            commonProperties,
+            inheritedProperties
+          )
+          tables += (tableUrl.asText -> table)
+        }
+        case _ => {
+          warnings = warnings :+ ErrorMessage(
+            "invalid_table_description",
+            "metadata",
+            "",
+            "",
+            tableElement.toPrettyString,
+            ""
+          )
+        }
+      }
+    }
+    (tables, warnings)
+  }
+
+  private def getId(commonProperties: Map[String, JsonNode]) = {
+    commonProperties
+      .get("@id")
+      .map(idNode => idNode.asText())
+  }
+
+  def getNotes(commonProperties: Map[String, JsonNode]): Option[JsonNode] = {
+    commonProperties.get("notes")
```

Comment:
Is this method really necessary?

Reply:
Initially that method had few lines of code. After many rounds of refactoring it came down to just a line. Ya I can do it inline now and I have removed this method